### PR TITLE
[optim] refactor adamax to use functional API

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56185 [optim] take kw-only argument for functional optim APIs
* #55834 [dist_optim] add distributed functional rprop optimizer
* #55833 [dist_optim] Add distributed functional Adamax optimizer
* #55832 [optim] refactor rprop to use functional API
* **#55830 [optim] refactor adamax to use functional API**

Differential Revision: [D26561017](https://our.internmc.facebook.com/intern/diff/D26561017/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26561017/)!

### CODE REVIEW

File: torch/optim/_functional.py [hunk 1]
```diff
@@ -238,3 +238,42 @@ def rmsprop(params: List[Tensor],
...
             param.addcdiv_(grad, avg, value=-lr)
+
+
+def adamax(params: List[Tensor],
+           grads: List[Tensor],
+           exp_avgs: List[Tensor],
+           exp_infs: List[Tensor],
+           state_steps: List[int],
+           eps: float,
+           beta1: float,
```
Comment (albanD) [inline torch/optim/_functional.py:249]: Could we make these keyword only arguments? That will reduce the chance to mix up the order of these in the function call.

**Review by albanD — APPROVED** [review]
Looks good.
What do you think about the change to kwargs? (similar idea for the other functional APIs btw)

File: torch/optim/_functional.py (hunk 1 above)
Comment (wanchaol) [reply torch/optim/_functional.py:249]: Yeah that make sense, let me create a follow up PR to change that.

Comment (wanchaol) [reply torch/optim/_functional.py:249]: Here is the PR: https://github.com/pytorch/pytorch/pull/56185

**Pull request closed without merging**